plugins/HHAnalyzer.cc

```diff
@@ -600,6 +600,21 @@ void HHAnalyzer::analyze(const edm::Event& event, const edm::EventSetup&, const
             myllmetjj.melaAngles = getMELAAngles(llmet[illmet].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4);
             myllmetjj.visMelaAngles = getMELAAngles(ll[ill].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4); // only take the visible part of the H(ww) candidate
 
+            // Compute MT2. See https://arxiv.org/pdf/1309.6318v1.pdf and https://arxiv.org/pdf/1411.4312v5.pdf
+            constexpr double M_B = 4.66;
+            constexpr double M_W = 80.385;
+            double px_invisible = myllmetjj.lep1_p4.px() + myllmetjj.lep2_p4.px() + myllmetjj.met_p4.px();
+            double py_invisible = myllmetjj.lep1_p4.py() + myllmetjj.lep2_p4.py() + myllmetjj.met_p4.py();
+
+            myllmetjj.MT2 = asymm_mt2_lester_bisect::get_mT2(
+                    M_B, myllmetjj.jet1_p4.px(), myllmetjj.jet1_p4.py(),
+                    M_B, myllmetjj.jet2_p4.px(), myllmetjj.jet2_p4.py(),
+                    px_invisible, py_invisible,
+                    M_W, M_W,
+                    1 // Absolute precision
+                    );
```

Comment:
Let's try that. Concerning the precision, I do not get your argument? If I understood correctly, it's indeed the absolute precision on MT2, ie `MT2 +/- X`. 0 means machine precision, but it's really slow. I don't think we are precise enough to go below 1 GeV though.